cylc/flow/network/scan.py

```diff
@@ -316,10 +316,21 @@ async def is_active(flow, is_active):
             False to filter for stopped and unregistered flows.
 
     """
-    contact = flow['path'] / SERVICE / CONTACT
-    _is_active = contact.exists()
+    service = flow['path'] / SERVICE
+    # NOTE: We must list the service directory contents rather than checking
+    # for the existence of the contact file directly, because listing the
+    # directory forces NFS filesystems to recompute their local cache.
+    # See https://github.com/cylc/cylc-flow/issues/6506
+    try:
+        contents = await scandir(service)
+    except FileNotFoundError:
+        _is_active = False
+    else:
+        _is_active = any(
+            path.name == WorkflowFiles.Service.CONTACT for path in contents
+        )
```

Comment:
Is this iteration needed as part of the listing, or could it be instead?
```suggestion
        _is_active = (service / CONTACT).exists()
```

Reply:
Yes, this is the point of the PR.

Reply:
We can either iterate (compute cost) or go back to the filesystem (FS cost).

There are only a couple of items in the directory, and the contact file will probs be the first (alpha order) so this is going to be faster.

Reply:
```suggestion
        _is_active = CONTACT in contents
```